service/history/reset/resetter.go

```diff
@@ -202,32 +202,58 @@ func (r *workflowResetterImpl) prepareResetWorkflow(
 		return nil, err
 	}
 
-	// TODO add checking of reset until event ID == decision task started ID + 1
 	decision, ok := resetMutableState.GetInFlightDecision()
-	if !ok || decision.StartedID+1 != resetMutableState.GetNextEventID() {
-		return nil, &shared.BadRequestError{
-			Message: fmt.Sprintf("Can only reset workflow to DecisionTaskStarted + 1: %v", baseRebuildLastEventID+1),
+	if ok {
+		// reset workflow has decision task start
+		if decision.StartedID+1 != resetMutableState.GetNextEventID() {
+			return nil, &shared.BadRequestError{
+				Message: fmt.Sprintf("Can only reset workflow to DecisionTaskStarted + 1: %v", baseRebuildLastEventID+1),
```

Comment:
This error is no longer a bad request. ResetMutableState.GetNextEventID() must be equal to decision.StartedID+1. This is an internal mutableState corruption if not true.

Reply:
removed